src/lib/github.js

```diff
@@ -119,6 +120,27 @@ async function fetchReposForProfile (profile, accessToken) {
   return accessToken ? repos : publicRepos;
 }
 
+function searchFilesFromRepo (repo, searchPattern, accessToken) {
```

Comment:
I'm not sure if we can leverage `octokit` to execute this query as opposed to re-implementing it here.

Reply:
We can, eg: `fetchWithOctokit('search.code', params, accessToken)`. Let's do it.

Reply:
Updated - this makes it much cleaner!